### PR TITLE
Node revamp: Environment Variables lesson

### LINKED ISSUE #27907: New Lesson: Environment Variables
## Lesson overview

What and why env vars && using `dotenv`. This new lesson will be placed before the new Express section. The [security configuration lesson](https://www.theodinproject.com/lessons/nodejs-security-configuration) will be removed as a result.

## Lesson outline

The following items define the scope of the lesson. The lesson outline is flexible; you can improve, expand, and omit items whilst writing lesson content (make sure the pull request description mentions that).

- What are environment variables and why they're useful && env vars live in Node's `process.env`
- Mention that we can manually populate `process.env` using the `export` cli command
- Mention there's a better way with the `dotenv` package
- Give a simple code example of installing & using `dotenv`

## Acceptance criteria

If the requirements here are not met, the work effort is not complete.

- [ ] Lesson conforms to the outline above (unless outline has been modified).
- [ ] Lesson follows our [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md).

### Get the lesson template

Download the lesson template using the following command (replace `lesson_name` with the actual lesson name):

```bash
curl -o <lesson_name>.md https://raw.githubusercontent.com/TheOdinProject/curriculum/main/templates/lesson-template.md 
```

### PR DESCRIPTION
**Pull request by MaoShizhong**
## Because

Node revamp env lesson, to be placed somewhere before "Introduction to Express" and to replace the existing "Security Configuration" lesson.

## This PR

- Adds new Node lesson on Environment Variables

## Issue

Closes #27907

## Additional Information

Kept `dotenv` instead of the still-experimental `--env-file` flag.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Review by cmedina-dev — COMMENTED** [review]
Honestly this looks great overall and I don't really have any recommendations for changes. The only addition I might suggest is a small blurb on the common pattern of `process.env.X || <default value>`. This looks great though!

**Comment by MaoShizhong** [comment]
The `env || default` pattern I didn't write anything about because I wanted to focus on what environment variables are and when/why we might want to use them. The idea being if you know what something is, realising when you might want `||` or `??` for a specific situation would be up to you.

The Intro to Express lesson (which will come at some point after this one) does demonstrate `const port = process.env.PORT || "3000"` though.

File: nodeJS/express/environment_variables.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+### Introduction
+
+When you run your code, your code runs in an individual environment. Run your code on a different machine? Different environment. Host your website somewhere like Netlify or Vercel? Different environment to your machine. Each environment is a unique collection of tools and configurations.
```
Comment (01zulfi) [inline nodeJS/express/environment_variables.md:3]: I'm not sure if "individual environment" is the right phrasing here. Maybe "specific environment" or "defined environment".

File: nodeJS/express/environment_variables.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+- Provide different values for different environments, such as your machine during development and a website host when deployed, without having to modify the source code.
+- Store secrets, such as database URLs and credentials, or API keys.
+
+For example, a production environment may want additional logging and analytics compared to a development environment. The source code may therefore have an environment variable for whether it's running in "dev" or "prod" mode and do different things based on that.
+
+Or perhaps you're building an API connected to a database, but you want to use a separate test database during development instead of the production one. You can pass your test database's URL and credentials into your app when you develop locally, but the deployment will have the values for the production database. On top of this, you can store environment variable values in a file that you add to your `.gitignore`, preventing the contents of that file from being exposed when changes are pushed.
+
+### Loading environment variables
```
Comment (01zulfi) [inline nodeJS/express/environment_variables.md:26]: I think we can add a short mention of the `export` cli command.

File: nodeJS/express/environment_variables.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+There are multiple ways you can load environment variables, though some are more cumbersome or may not have stable support yet in many Node versions. One way is by defining the environment variables and their values directly in the command to run your code. Instead of running your app with just `node index.js`, you could run the following (note that quotes are optional for values that do not contain certain special characters like spaces or `=`):
+
+```bash
+MODE=prod VIDEO_URL="https://www.youtube.com/watch?v=X2CYWg9-2N0" node index.js
```
Comment (01zulfi) [inline nodeJS/express/environment_variables.md:31]: lmao

**Review by 01zulfi — CHANGES_REQUESTED** [review]

File: nodeJS/express/environment_variables.md (hunk 1 above)
Comment (MaoShizhong) [reply nodeJS/express/environment_variables.md:3]: "specific" probably feels a bit smoother here :+1:

File: nodeJS/express/environment_variables.md
Comment (01zulfi) [inline nodeJS/express/environment_variables.md:1]: also, can we move this to the `introduction_to_nodeJS` folder as this lesson will be placed in that section. or, are you thinking of keeping the lesson in this new `express` section?

Comment (MaoShizhong) [reply nodeJS/express/environment_variables.md:1]: Oh. Oops.
Thanks for catching that!

**Review by 01zulfi — APPROVED** [review]
great stuff 🔥 

merge at your pleasure @MaoShizhong 🚀

**Review by 01zulfi — APPROVED** [review]

**Pull request merged**